### ISSUE TITLE
About search interative and sensibility

### ISSUE THREAD

**Issue by acpguedes** [soedinglab/MMseqs2#106]
I wondering to know a simple thing about search tool.  
Since I read the [mmseqs paper](https://academic.oup.com/bioinformatics/article/32/9/1323/1744460) I was too interested and start to use it about 1 year with really good results.  
Recently I start to use to iterative searches, since the [wiki](https://github.com/soedinglab/MMseqs2/wiki#set-sensitivity--s-parameter) report higher sensitivity of MMseqs compared with PSI-Blast as reported by this graph.  

I wasn't able to reproduce this results, first because I have no access to dataset used to do it and at least I didn't find the parameters used in MMseqs2profile.  

So I am just trying to do some different tests against different databases (NCBI nr, PDB, Uniprot50 20 ...) but I yet didn't find any parameter which de sensitivity of MMseqs with 3 or 4 iterations are higher than PSI-Blast.   
So my questions are:  
__Which are the best parameters on MMseqs search keep closer or better than default parameters used on PSI-Blast (NCBI reference)?__

Thanks in advance

**Comment by martin-steinegger** [comment]
Thank you for trying MMseqs2.  

Your referenced article is the MMseqs1 paper. Please read the MMseqs2 [article](https://www.nature.com/articles/nbt.3988). The benchmark used to evaluate MMseqs2 is freely available [here](https://bitbucket.org/martin_steinegger/mmseqs-benchmark). 

How did you compare the sensitivity of MMseqs2 against PSI-BLAST? We measured the sensitivity up to the first false positive. Our benchmark penalizes high scoring FPs.

**Comment by acpguedes** [comment]
I'm trying to do more precise comparisons against both control the parameters of both to same results size, inclusion e-value for profile composition, etc... (closer the parameters you use in benchmark)

I think, in next versions you could include controls to calculate score in smith-waterman step, this way the user can control gap-opening and extension and others parameters and also others compositional bias correction. But I made it work with high sensitivity compared to psiblast. I can report further. Thanks

**Comment by martin-steinegger** [comment]
Oh, great to hear that you could find parameters! I would be interested how you benchmark and what kind of parameters you used in the end.

What do you mean by controls? It would be easy to include gap open and extend costs to MMseqs2. But I think it would make more sense if the gap costs depends on the diversity of the profiles. More diverse profiles need lower gap open and extend costs compared to less divers once. So one global parameters might not fit all profiles equally. This is still an open point.

We do not explicitly encode the iterations in the results. But we append hits after each iteration to the previous results. So you could recognize iteration changes by comparing the e-value to previous hits. A lower e-values compared to the previous hits indicates a new iteration.

**Comment by acpguedes** [comment]
Hi @martin-steinegger , sorry for late response.

Well, I agree with you about gap costs and diversity of the profiles, but it is just to attend more flexibility of options, for example, commonly blast has fixed gap costs if the user really wants to use same gap costs it's possible. 

Well, I did not do anything so hard to benchmark, I'm send you the script I did, I also may send you the database, query and results, but I think the parameters I used is enough.

[scripts.zip](https://github.com/soedinglab/MMseqs2/files/2249981/scripts.zip)

 If you want all analysis just tell me and I send ASAP.

**Comment by acpguedes** [comment]
@martin-steinegger 
You told me that mmseqs appends results at each iteration, so I can recognize iteration just comparing  e-value to previous hit. I did it but doesn't work.

I run 16000 proteins against a Db with 17 millions, with this parameters:  
> mmseqs search query.db all.mmseqs found.db tmp/ -s 7 -a -e 5 --threads 15 --max-seqs 17000000 --alignment-mode 3 --num-iterations 3  

It run really fast, thanks for that.  

But when I try to check the e-values as your suggestion I found more than 3 iterations.

I'm send you an example extracted from results and without any treatment except the 'grep' to colect it witch suppose keep the results order as the original file.

> grep -wF AKM80458  tcdb_found.m8 > exemple.txt

[exemple.txt](https://github.com/soedinglab/MMseqs2/files/2280404/exemple.txt)

> grep -wF ^AKM80458  tcdb_found.m8 > exemple.mmseqs  

Fast  way to find iterations. I used this command:  
> perl -F'\t' -anE 'chomp @F; $F[10]=~s/E/e/; $F[12] = (!exists $H{$F[0]} || $H{e} > $F[10]) ? ++$H{$F[0]} : $H{$F[0]}; $H{e} = $F[10]; say join("\t", @F);' exemple.txt | less -S
